### DISCUSSION TITLE
Selected fuel tank indication

### DISCUSSION THREAD

**Discussion by conao68** [nguyenquyhy/Flight-Tracker-StreamDeck#244] [Q&A]
What a great plugin this is! Thank you very much for your hard work! I just bought the Stream Deck XL and I'm having fun configuring my first aircraft, the X Cub. I got almost everything working as I wanted, but I have a few questions/issues:

1. I can select the fuel tanks, using FUEL_SELECTOR_LEFT/FUEL_SELECTOR_RIGHT, but I cannot find a way to indicate that the respective tank is selected. I tried FUEL_TANK_SELECTOR__1/FUEL_TANK_SELECTOR__2 and FUEL_TANK_SELECTOR==2/FUEL_TANK_SELECTOR==3 (this variable exist in the SDK documentation but not in this plugin's list, so maybe it should not work anyway). Anyone got this to work? How?

2. Same issue with the IGN LH and IGN RH. The MAGNETO1_LEFT/MAGNETO1_RIGHT events work to toggle them, but I could not find a variable that could indicate it. Any ideas?

3. Is there a way to write a value to a variable using the Generic Toggle Button? For instance, if I want to set the rudder trim to 0 with a button press, I thought I could use the toggle event ELEVATOR_TRIM_SET and the toggle parameter ELEVATOR_TRIM_POSITION=0, but I guess this doesn't work. If I understood the documentation right, I can either pass a value or a variable, not both.

I'm sure I'll have more questions, but this is enough for a start.  :-)

**Comment by peteclyde** [comment]
Point number 1 above is exactly what I came here looking for - I'm at a loss so far.

Did anyone ever get that solved?  ie: a value indicating which fuel tank was in use ?

Thanks for any assistance.

Pete M

**Reply by nguyenquyhy** [reply]
Have you tried `FUEL TANK SELECTOR:1==2` for left tank and `FUEL TANK SELECTOR:1==3` for right tank? I tried it in the Bonaza and it works fine for me.

**Comment by beckerdo** [comment]
I would love to see an answer to this question also. I have toggles for my Stream Deck which properly set the fuel selector using the FUEL_SELECTOR_LEFT and FUEL_SELECTOR_RIGHT toggle values. But I am at a loss as to what feedback value to use to change the Generic Toggle button from red to green.

I see this page https://docs.flightsimulator.com/html/index.htm#t=Programming_Tools%2FSimVars%2FAircraft_SimVars%2FAircraft_Fuel_Variables.htm%23FUELSYSTEM_LINE_FUEL_FLOW but I have trouble translating the Microsoft Values into something that Flight Tracker can use.

Thanks, Dan

**Reply by nguyenquyhy** [reply]
Have you tried `FUEL TANK SELECTOR:1==2` for left tank and `FUEL TANK SELECTOR:1==3` for right tank? I tried it in the Bonaza and it works fine for me.

**Reply by beckerdo** [reply]
Thanks nguyenquyhy! Your instructions of FUEL TANK SELECTOR:1==2 and FUEL TANK SELECTOR:1==3 worked perfectly for me. When pressing the button for FUEL_SELECTOR_RIGHT, not only did the FUEL L tank button turn red, but the FUEL R tank button turned green. This saves me the time of fidgeting with the mouse, or changing my view to look at the tank selector on the floor.

Also, this is the first time I saw the use of one of the new SDK sim variables with an index value. I'm sure I will use this pattern with other controls.

**Comment by beckerdo** [comment]
Another question related to my FUEL_SELECTOR questions. I see in the Stream Deck app on Windows (version 5.2.0.14948) that there are icons associated with my fuel selector.

The main button has a tiny drop down with Icon/Set from File/Create New Icon/Open Stream Deck Icon Library. The Custom ON/OFF images: section has ON and OFF images with Embed/Link/Clear buttons with names like "(Embed) park on.png".

Where are these images located? Anyway to extract embeded images? Currently fuel selector off shows as green and on shows as red. I would like to edit and change this.

**Reply by nguyenquyhy** [reply]
If it says embedded, it is encoded inside the profile itself so it is easier for profile sharing. Unfortunately I don't think there is any easy way to extract that yet.

If you simply want to flip ON/OFF image, maybe you can add `==0` to the end of feedback value to flip the meaning of ON and OFF.

**Reply by beckerdo** [reply]
The other thing I did was take a screen capture of the embedded button image in the Stream Deck app, edited in photo editor, saved it, cleared the old button image out, and imported the new image into the profile.

Thanks for all your timely help! Dan
